src/state/apis/swapper/helpers/validateTradeQuote.ts

```diff
@@ -221,9 +222,22 @@ export const validateTradeQuote = async (
     // Swappers other than THORChain shouldn't be affected by this limitation
     if (swapperName !== SwapperName.Thorchain) return false
 
-    // Sender is either a smart contract address, or the bytecode is still loading - disable confirm
-    const _isSmartContractAddress = await isSmartContractAddress(sendAddress)
-    if (_isSmartContractAddress !== false) return true
+    // This is either a smart contract address, or the bytecode is still loading - disable confirm
+    const _isSmartContractSellAddress = await isSmartContractAddress(sendAddress)
+    const _isSmartContractReceiveAddress = await isSmartContractAddress(quote.receiveAddress)
+    // For long-tails, the *destination* address cannot be a smart contract
+    // https://dev.thorchain.org/aggregators/aggregator-overview.html#admonition-warning
+    // This doesn't apply to regular THOR swaps however, which docs have no mention of *destination* having to be an EOA
+    // https://dev.thorchain.org/protocol-development/chain-clients/evm-chains.html?search=smart%20contract
+    if (
+      firstHop.source === THORCHAIN_LONGTAIL_STREAMING_SWAP_SOURCE &&
```

Comment:
Need so check non-streaming longtails, also eventually we might support longtail out, so we can do the following:

```suggestion
      [firstHop.source , secondHop.source].some(source => [THORCHAIN_LONGTAIL_SWAP_SOURCE, THORCHAIN_LONGTAIL_STREAMING_SWAP_SOURCE].includes(source)) &&
```

Reply:
gm 8daca484a3